src/main/java/ws/wamp/jawampa/WampClient.java

```diff
@@ -1107,14 +1171,20 @@ public void run() {
                         }
                         else { // need to subscribe
                             // Insert a new entry in the subscription map
-                            final SubscriptionMapEntry newEntry = new SubscriptionMapEntry(PubSubState.Subscribing);
+                            final SubscriptionMapEntry newEntry = new SubscriptionMapEntry(flags, PubSubState.Subscribing);
                             newEntry.subscribers.add(subscriber);
-                            subscriptionsByUri.put(topic, newEntry);
+                            subscriptionsByFlags.get(flags).put(topic, newEntry);
 
                             // Make the subscribe call
                             final long requestId = IdGenerator.newLinearId(lastRequestId, requestMap);
                             lastRequestId = requestId;
-                            final SubscribeMessage msg = new SubscribeMessage(requestId, null, topic);
+
+                            ObjectNode options = null;
+                            if (flags != SubscriptionFlags.Exact) {
+                                options = new ObjectMapper().createObjectNode();
```

Comment:
Should use the objectMapper field of the client class instead of creating a new one